### ISSUE TITLE
[Bug]: Example in documentation for state management using JS Objects is not working

### ISSUE THREAD

**Issue by ItsiW** [appsmithorg/appsmith#22881]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I attempt to try Example 1 for storing state using JS Objects described in [the documentation](https://docs.appsmith.com/core-concepts/writing-code/javascript-editor-beta), I am not getting described results. I have had to resort to using the appsmith.store instead of local variables to use state in my apps.

Trying the example, the state does not get modified, and the text to display the counter stays at 1. 

I added logging
```
export default {
        counter: 1,
        increment: () => {
	    console.log("Original counter: " + this.counter);
            this.counter = this.counter + 1;
	    console.log("Current counter: " + this.counter)
        },
        decrement: () => {
            this.counter = this.counter - 1
        }
}
```

and my logs show `Original counter: 1, Current counter: 2`, every time I hit the increment button. So the button is working, but the state is not being modified. The text also does not change.

### Steps To Reproduce

1. Create the JSObject1 with counter, increment, and decrement, as well as the text and button widgets as described in Example 1 of "Use variables for state management" in https://docs.appsmith.com/core-concepts/writing-code/javascript-editor-beta
2. Clicking the buttons does not change the state

### Environment

Deploy Preview

### Version

Appsmith Community v1.9.12

**Comment by dbrcplex** [comment]
noticed this today as well. no new info to add but good to know it is on the radar as this feature would be nice if it is working.

**Comment by rhys-phelps-labs** [comment]
Would like to up vote this as well. This is causing a challenge when trying to use supabase-js client. Since the supabase client is non-serializable it cannot be in the store. Creating the problem of not having it in any memory location.

**Comment by jnikhila** [comment]
@bharath31 - Could you please help us here? It would be really great if you could raise a PR for the change and we can take it ahead from there. Thanks.

**Comment by bharath31** [comment]
@ItsiW @rhys-phelps-labs @dbrcplex I'm unable to reproduce this issue. Could you confirm if this is resolved?

**Comment by rhys-phelps-labs** [comment]
Hi Bharanth,

No, still not working.

https://app.picniclabs.io/app/testing/test-64b72726f049835c16d51efa/edit/jsObjects/64e5a8adf049835c16d524ff

I can keep this open for a while, so you can test if you like.

Appsmith Community v1.9.15
Chrome: Version 116.0.5845.97 (Official Build) (64-bit)
Windows 11

_________________________________

Rhys Phelps
Technical Specialist
M: +614 3864 0701

**Comment by bharath31** [comment]
@rhys-phelps-labs I'm unable to access your app. Could you please share a loom video instead?

**Comment by rhys-phelps-labs** [comment]
Hi Bharath,

Sorry about the access issue.

Here is the loom recording.

https://www.loom.com/share/977850dc80ec44db8fd75e3b690e3784?sid=574be26f-df9f-4144-b4e7-8b12eb75f8ac

Regards

_________________________________

Rhys Phelps
Technical Specialist
M: +614 3864 0701

**Comment by Rishabh-Rathod** [comment]
@rhys-phelps-labs @ItsiW @dbrcplex Could you please help us reproduce this issue by inviting `support@appsmith.com` as a developer to the development version of the app and sharing the link to it? 

You could email the link to us at `support@appsmith.com` 
PS: please cc `rishabh.rathod@appsmith.com` .

**Comment by rhys-phelps-labs** [comment]
https://app.picniclabs.io/app/testing/test-64b72726f049835c16d51efa


***@***.***
_________________________________

Rhys Phelps
CTO
1300 687 275
M: +614 3864 0701

**Comment by Rishabh-Rathod** [comment]
@rhys-phelps-labs JSObject variable mutation feature was released in appsmith version [1.9.16](https://github.com/appsmithorg/appsmith/releases/tag/v1.9.16) and shared instance appsmith version is 1.9.15. 

Please update the version to use the above feature. 

I am closing this issue for now, please reopen if after update the problem is not resolved.
